### PR TITLE
Feat/improve scanning

### PR DESCRIPTION
**Pull request by edge33**
### Ticket:

- [User Story 18798](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18798): improve anti virus scanning

### Description

- Improved and optimised av scanning

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

**Review by David-Losinski — COMMENTED** [review]
Tests look good to me

File: apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE files
+ADD COLUMN antivirus_db_version VARCHAR(255) DEFAULT NULL;
```
Comment (Thurfjell) [inline apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql:2]: text?

File: apps/upload-api/.eslintrc.json [hunk 2]
```diff
@@ -11,11 +11,22 @@
...
     "browser": true,
     "node": true
   },
-  "ignorePatterns": ["dist", "db"]
-}
+  "ignorePatterns": [
+    "dist",
+    "db"
+  ]
+}
```
Comment (peschina) [inline apps/upload-api/.eslintrc.json:32]: Can you please add EOF?

File: package-lock.json
Comment (peschina) [inline package-lock.json:1]: Are the changes here expected?

**Review by Thurfjell — COMMENTED** [review]
Are we selecting deleted files in the get requests? Is this intentional?

File: apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql (hunk 1 above)
Comment (edge33) [reply apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql:2]: it is just going to be a string, wouldn't use a text field here

**Review by peschina — DISMISSED** [review]
LGTM, a few minor comments

File: package-lock.json
Comment (edge33) [reply package-lock.json:1]: I removed some unused packages and these changes might be caused by it

**Review by peschina — DISMISSED** [review]

File: apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql (hunk 1 above)
Comment (Thurfjell) [reply apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql:2]: Stored the same regardless. `text` looks nice and default. Do as you want.

Comment (edge33) [reply apps/upload-api/db/migrations/sqls/20240809142154-add-antivirus-db-field-up.sql:2]: ah, just learnt that, thanks for sharing, I agree on using text

**Review by Thurfjell — APPROVED** [review]

**Pull request merged**